src/EFCore.PG.NodaTime/Storage/Internal/NpgsqlNodaTimeTypeMappingSourcePlugin.cs

```diff
@@ -12,13 +12,22 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal
 {
     public class NpgsqlNodaTimeTypeMappingSourcePlugin : IRelationalTypeMappingSourcePlugin
     {
+#if DEBUG
+        internal static bool LegacyTimestampBehavior;
+#else
+        internal static readonly bool LegacyTimestampBehavior;
+#endif
+
+        static NpgsqlNodaTimeTypeMappingSourcePlugin()
+            => LegacyTimestampBehavior = AppContext.TryGetSwitch("Npgsql.EnableLegacyTimestampBehavior", out var enabled) && enabled;
```

Comment:
Technically true, but it's very very hard to imagine a non-intentional scenario where this can happen... In general, an application flipping around an appcontext switch is not a normal scenario, and I suspect various other .NET components check switches in multiple parts of the codebase rather than capturing it once...

Do you think we should make the flag a public API in Npgsql? Seems a bit much to me...